### ISSUE TITLE
[BUG] Cannot create a Managed application definition

### ISSUE THREAD

**Issue by sergiygetlin** [Azure/azure-libraries-for-net#1279]
**Describe the bug**
I am trying to create a Managed Application Definition. My request failed because of "InvalidRequestContent" exception. I turned on logging and noticed that the body of the request doesn't include 'properties'. Code snippet and screenshot of the request details are attached. Please find them below.

***Exception or Stack Trace***
InvalidRequestContent due to missed data in the request's body

**To Reproduce**
Please try code snippet below.

***Code Snippet***
    `

     var managedAppDefProperties =
          new
          {
                lockLevel = "ReadOnly",
                displayName = "test",
                description = "test",
                packageFileUri = packageFileUri,
                authorizations = new object[] { new { principalId = principalId, roleDefinitionId = roleDefinitionId } }
          };
    azureClient.GenericResources
          .Define(managedAppDefName)
          .WithRegion(region)
          .WithExistingResourceGroup(existingResourceGroup)
          .WithResourceType("applicationDefinitions")
          .WithProviderNamespace("Microsoft.Solutions")
          .WithoutPlan()
          .WithApiVersion("2019-07-01")
          .WithProperties(managedAppDefProperties)
          .WithTag("key", "value")
          .Create();
`

**Expected behavior**
Request body should contain defined properties

**Screenshots**
<img width="652" alt="api_issue" src="https://user-images.githubusercontent.com/91149690/137031243-65741141-e4fc-4322-ba39-63ad2f2e9bd1.png">

**Setup (please complete the following information):**
 - OS: Windows 10
 - IDE : Visual Studio 2019
 - Version of the Library used: 1.38.0

**Comment by weidongxu-microsoft** [comment]
@sergiygetlin 

The object to `withProperties` is to be serialized to JSON.

I am not sure what type will be created via the syntax in your code. Could you try `IDictionary`?

**Comment by weidongxu-microsoft** [comment]
Also new SDK is here https://aka.ms/azsdk/dotnet/mgmt

**Comment by sergiygetlin** [comment]
Thank you @weidongxu-microsoft. I checked new SDK and they (old and new) definitely uses different libs for serialization. The old one can accept IDictionary<object, object>, but the new one require it to be at least IDictionary<string, object>. So, the key should be string. It looks like it is requirement of serialization lib.

I have one more question about old lib... I noticed that in fluent library there is only one method "WithProperties" which sets additional data. But for instance in case we need to create managed application (not definition) it requires additional parameters such as "kind" which should be presented in the request body. So the body will be something like this
```
{
 "kind" : "ServiceCatalog",
 "properties" : 
    {
    }
}
```
If we try to set "kind" using "WithProperties" it will be included into "properties" which is not correct. And it looks like there is no other way of setting up that parameter. Did I miss something?
Thank you.

**Comment by weidongxu-microsoft** [comment]
@sergiygetlin 

Yes, this is a flaw of the old lib. It should have a `withKind` for you to set it.

The new lib should work at this case.
